packages/ionic/src/commands/deploy/manifest.ts

```diff
@@ -31,10 +32,24 @@ export class DeployManifestCommand extends Command {
       throw new FatalException(`Cannot run ${input('ionic deploy manifest')} outside a project directory.`);
     }
 
-    const buildDir = path.resolve(this.project.directory, 'www'); // TODO: this is hard-coded
+    const integration = await this.getAppIntegration();
+    let buildDir: string;
+    if (integration && ['capacitor', 'cordova'].includes(integration)) {
+      buildDir = await this.project.getDistDir();
+    } else {
+      throw new FatalException(
```

Comment:
This message is still written in two places. Why can't you reuse it from `deploy/core.ts`?